keps/3122-expose-flavors-in-localqueue-status/README.md

```diff
@@ -0,0 +1,141 @@
+# KEP-3122: Expose Flavors in LocalQueue Status
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+This KEP introduces a new status field in LocalQueue, allowing users to see
+all currently available ResourceFlavors in the LocalQueue.
+
+## Motivation
+
+Currently, users without RBAC access to ResourceFlavors cannot view the list
+of available flavors. Depending on the RBAC rules, users might also lack read
+access to ClusterQueues. Providing users with information about the available
+flavors is useful, as it gives them an idea of the capabilities provided by
+a LocalQueue (e.g., a flavor might include newer GPUs).
+
+### Goals
+
+- Provide a possibility to see all currently available ResourceFlavors in 
+  the LocalQueue.
+
+### Non-Goals
+
+- Verify that the ResourceFlavors exist and show only the existing flavors.
+
+## Proposal
+
+Introduce a new status field `availableFlavors` in LocalQueue 
+that will be updated when ClusterQueue flavors are modified.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a user I want to see the list of all ResourceFlavors available in each LocalQueue 
+due to the RBAC configuration for ClusterQueue in my cluster I cannot inspect the 
+ClusterQueue objects directly, only LocalQueues.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+## Design Details
+
+Modify `LocalQueueStatus` API object:
+
+```
+// LocalQueueStatus defines the observed state of LocalQueue
+type LocalQueueStatus struct {
+	[...]
+	
+	// availableFlavors lists all currently available ResourceFlavors
+	// in specified ClusterQueue.
+	//
+	// +listType=set
+	// +optional
+	AvailableFlavors []ResourceFlavorReference `json:"availableFlavors,omitempty"`
```

Comment:
I'm wondering if maybe instead of list of strings return a list of objects, with just `name` currently, like this:
```yaml
availableFlavors:
  - name: on-demand
  - name: spot
```

To make the model more flexible, I imagine in the future we may want to return also the list of covered resources per resource flavor, or if the flavor exists (see non-goals). WDYT @tenzen-y @alculquicondor ?

Reply:
Listing covered resources would be a welcome extension

Reply:
Ok, cool, so the question remains how much information would be useful. We could have just `coveredResources: []string` for the start, but also with quantities, like here:
```yaml
      resources:
      - name: "cpu"
        nominalQuota: 9
        borrowingLimit: 12
      - name: "memory"
        nominalQuota: 36Gi
```
Any opinions @KPostOffice @varshaprasad96 @alculquicondor @tenzen-y ?

Reply:
> ```yaml
>  resources:
>       - name: "cpu"
>         nominalQuota: 9
>         borrowingLimit: 12
>       - name: "memory"
>         nominalQuota: 36Gi
> ```

Done

Reply:
> I'm wondering if maybe instead of list of strings return a list of objects, with just `name` currently, like this:
> 
> ```yaml
> availableFlavors:
>   - name: on-demand
>   - name: spot
> ```
> 
> To make the model more flexible, I imagine in the future we may want to return also the list of covered resources per resource flavor, or if the flavor exists (see non-goals). WDYT @tenzen-y @alculquicondor ?

Done